### DISCUSSION TITLE
table attributes

### DISCUSSION THREAD

**Discussion by hudsondiegof** [squidfunk/mkdocs-material#6173] [Q&A] [answered]
Is there a way to customize tables with attributes? Keeping the table in markdown format, but referencing a custom style? Something like:

```
| Header 1 | Header 2 |
|----------|----------|
| Cell 1   | Cell 2   |
| Cell 3   | Cell 4   |
{:.fancy-table}

```
```
| Header 1 | Header 2 |
|----------|----------|
| Cell 1   | Cell 2   |
| Cell 3   | Cell 4   |
{:.simple-table}
```
```
{: .simple-table}
| Header 1 | Header 2 |
|----------|----------|
| Cell 1   | Cell 2   |
| Cell 3   | Cell 4   |
```
```
::: table-class-2
| Cabeçalho 1 | Cabeçalho 2 |
|-------------|-------------|
| Conteúdo   | Conteúdo   |
:::
```
My project has multiple styles of tables, i don't want to use html to maintain the readability of the .md file.

**Comment by facelessuser** [comment, marked as answer]
Unfortunately, `attr_list` won't work with tables. One possibility is using the [`pymdownx.blocks.html`](https://facelessuser.github.io/pymdown-extensions/extensions/blocks/plugins/html/) extension. That is if you find the syntax more attractive than raw HTML

The below example wraps the table in a `div` with the specified class:

```py
import markdown

md = """
/// html | div.class
col1  | col2
----- | ----
cell1 | cell2
///
"""

print(markdown.markdown(md, extensions=['tables', 'pymdownx.blocks.html']))
```

```html
<div class="class">
<table>
<thead>
<tr>
<th>col1</th>
<th>col2</th>
</tr>
</thead>
<tbody>
<tr>
<td>cell1</td>
<td>cell2</td>
</tr>
</tbody>
</table>
</div>
```

**Comment by hudsondiegof** [comment]
Yes, it's very close to what I need.

Thanks.


Code:
![image](https://github.com/squidfunk/mkdocs-material/assets/83621099/90cbea32-2ca5-44e7-bd88-1894ccd30fec)

Output in mkdocs material:
![image](https://github.com/squidfunk/mkdocs-material/assets/83621099/f0e6f2d4-84a5-44fe-8840-e293a7fb0fd9)

I added these extensions to mkdocs.yml
![image](https://github.com/squidfunk/mkdocs-material/assets/83621099/db0ef1cf-a18b-4b8f-bd4d-5e21b8ea63b7)

**Reply by facelessuser** [reply]
Glad it worked out for you 🙂.
